index.js

```diff
@@ -290,19 +322,13 @@ export default class OneSignal {
     static setEmail(email, emailAuthCode, callback) {
         if (!checkIfInitialized()) return;
 
-        if (emailAuthCode == undefined) {
-            //emailAuthCode is an optional parameter
-            //since JS does not support function overloading,
-            //unauthenticated setEmail calls will have emailAuthCode as the callback
+        if (emailAuthCode === undefined)
+            emailAuthCode = '';
```

Comment:
`emailAuthCode` should be either `null` or `undefined` if it isn't used. Otherwise I believe the native SDK will try to send a blank string to the backend and reject it.

Reply:
oh thank you sir